src/server/pkg/log/log.go

```diff
@@ -26,16 +30,108 @@ func NewLogger(service string) Logger {
 	l.Formatter = new(prettyFormatter)
 	return &logger{
 		l.WithFields(logrus.Fields{"service": service}),
+		make(map[string]*prometheus.HistogramVec),
+		make(map[string]prometheus.Counter),
 	}
 }
 
 // Helper function used to log requests and responses from our GRPC method
 // implementations
 func (l *logger) Log(request interface{}, response interface{}, err error, duration time.Duration) {
+	state := "started"
 	if err != nil {
 		l.LogAtLevelFromDepth(request, response, err, duration, logrus.ErrorLevel, 4)
+		state = "errored"
 	} else {
 		l.LogAtLevelFromDepth(request, response, err, duration, logrus.InfoLevel, 4)
+		if duration.Seconds() > 0 {
+			state = "finished"
+		}
+	}
+	l.ReportMetric(state, duration)
+}
+
+func (l *logger) ReportMetric(state string, duration time.Duration) {
+	depth := 4
+	pc := make([]uintptr, depth)
+	runtime.Callers(depth, pc)
+	split := strings.Split(runtime.FuncForPC(pc[0]).Name(), ".")
+	method := split[len(split)-1]
+	entry := l.WithFields(logrus.Fields{"method": method})
+
+	var tokens []string
+	for _, token := range camelcase.Split(method) {
+		tokens = append(tokens, strings.ToLower(token))
+	}
+	rootStatName := strings.Join(tokens, "_")
```

Comment:
Does Prometheus not accept camelcase names?

Reply:
It's certainly recommended to use [snake_case](https://prometheus.io/docs/practices/naming/), but it does seem to accept camel case ones. However, that means our names will be inconsistent ... e.g. `pachd_GetFile_count` since prometheus prepends the namespace and in some cases (e.g. histograms) appends the metric type (`count`, `sum`, or `bucket`). 

Also ... it'd be inconsistent between this and other stats. The worker stats we report are all snake case.

I get how adding a new dependency in here for handling camel case isn't ideal, but it seems worth it to have our metric names / queries be more consistent.

Reply:
Yeah, I had a similar thought, but I'm cool with it. `pachd_GetFile_count` is very ugly.